chatbot.py

```diff
@@ -143,7 +143,7 @@ def message_to_bot(H, clf, learn_response):
         B, learn_response = databaseconnect.learn_question_response(H)
     if (
         len(proper_nouns) >= 2
-        or (len(proper_nouns) >= 1 and H.split(" ", 1)[0] == "Where")
+        or (len(proper_nouns) >= 1 and H.split(" ", 1)[0] in ["Where", "What"])
```

Comment:
Checking the first word is poor practice. Since this block of code is executed only when the classification from the model is `question`, it's unnecessary.

Reply:
> Checking the first word is poor practice.

There is already a [check](https://github.com/vishakha-lall/MapBot/blob/gssoc-master/chatbot.py#L146) in place, for the first word "Where" and I decided to do it the same way and include the word "What". "Where" probably exists for checking `geocoding`, and was introduced in [this commit](https://github.com/vishakha-lall/MapBot/commit/7750dda9b500a33f3bca7fab69817f8e5e053a57#diff-70f0fc12d3b0ec546801eb20c528242eR81) 2 years ago. Do you then want me to remove the check for "Where" too?

> Since this block of code is executed only when the classification from the model is `question`, it's unnecessary.

I don't think that's true. `classification == 'Q'` is checked in [Line127](https://github.com/vishakha-lall/MapBot/blob/gssoc-master/chatbot.py#L127) whereas this line is indented such that it'd be executed without any dependency on the `classification`.

> The bot does not respond appropriately for questions like How high is Delhi above the sea level? or How high is Delhi?

Yeah. That is because it gets caught in [this mystery condition.](https://github.com/vishakha-lall/MapBot/blob/gssoc-master/chatbot.py#L134-L139) which doesn't perform any calls to the `googleMapsApiModule.py`. I'm unsure what this if statement is for and thus didn't remove it altogether. Though, including "high" as one of the keywords to check for, to execute elevation API, would do the trick and solve this.

> It looks like when the length of proper_noun is longer than 1, the response is not correct. If the location cannot be extracted correctly, the bot can ask a follow-up question to get the exact location from the user (As in case of destination for route)

I failed to think this through. I'll have a go and solve this ASAP.

Reply:
> > Checking the first word is poor practice.
> 
> There is already a [check](https://github.com/vishakha-lall/MapBot/blob/gssoc-master/chatbot.py#L146) in place, for the first word "Where" and I decided to do it the same way and include the word "What". "Where" probably exists for checking `geocoding`, and was introduced in [this commit](https://github.com/vishakha-lall/MapBot/commit/7750dda9b500a33f3bca7fab69817f8e5e053a57#diff-70f0fc12d3b0ec546801eb20c528242eR81) 2 years ago. Do you then want me to remove the check for "Where" too?
> 
> > Since this block of code is executed only when the classification from the model is `question`, it's unnecessary.
> 
> I don't think that's true. `classification == 'Q'` is checked in [Line127](https://github.com/vishakha-lall/MapBot/blob/gssoc-master/chatbot.py#L127) whereas this line is indented such that it'd be executed without any dependency on the `classification`.

I see the problem here, you can go ahead with the current approach in this issue, I will create a new issue to address this. It might require some code refactoring.

Reply:
So, I'll add another subject keyword: "**high**", and see to it that proper nouns of more than 1 words are recognized, right?